AI/Nullkiller/AIGateway.cpp

```diff
@@ -154,10 +154,13 @@ void AIGateway::artifactAssembled(const ArtifactLocation & al)
 	NET_EVENT_HANDLER;
 }
 
-void AIGateway::showTavernWindow(const CGObjectInstance * townOrTavern)
+void AIGateway::showTavernWindow(const CGObjectInstance * object, const CGHeroInstance * visitor, QueryID queryID)
 {
 	LOG_TRACE(logAi);
 	NET_EVENT_HANDLER;
+
+	status.addQuery(queryID, "TavernWindow");
+	requestActionASAP([=](){ answerQuery(queryID, 0); });
```

Comment:
I was thinking NKAI uses advmap taverns to hire heroes too

Reply:
Does not looks like it. From what I see NKAI can only recruit heroes from towns.

Reply:
Yep, I do not remember such functionality so expect that it can not